### PR TITLE
Fixed issue 18794 - My billing and shipping address are the same" checkbox won't be unchecked when multiple payment methods are available.

### PR DESCRIPTION
**Pull request by Dharmeshvaja91**
### Description (*)

My billing and shipping address are the same" checkbox won't be unchecked when multiple payment methods are available.

### Fixed Issues (if relevant)

1. magento/magento2#18794: My billing and shipping address are the same" checkbox won't be unchecked when multiple payment methods are available.

### Manual testing scenarios (*)

1.Add to cart a product, and go to checkout.
2.Select any address, and click "Next" button.
3.Select any payment method, and check "My billing and shipping address are the same
" checkbox.
4.Return to Shipping step.
5.Select other address, and click "Next" button. 

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @Dharmeshvaja91. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: app/code/Magento/Checkout/view/frontend/web/js/action/select-shipping-address.js [hunk 1]
```diff
@@ -7,11 +7,15 @@
...
-], function (quote) {
+    'Magento_Checkout/js/model/quote',
+    'Magento_Checkout/js/checkout-data',
+    'Magento_Checkout/js/action/select-billing-address'
+], function (quote, checkoutData) {
     'use strict';
-
+    
     return function (shippingAddress) {
+        window.isbothAddressSame = false;
```
Comment (orlangur) [inline app/code/Magento/Checkout/view/frontend/web/js/action/select-shipping-address.js]: Storing flags in global `window` object is not acceptable.

**Review by orlangur — CHANGES_REQUESTED** [review]

**Comment by orlangur** [comment]
@Dharmeshvaja91 a couple of functional tests are now broken, please check them out.

**Comment by Dharmeshvaja91** [comment]
> @Dharmeshvaja91 a couple of functional tests are now broken, please check them out.


@orlangur could you please check now i just updated my code.

**Comment by orlangur** [comment]
@Dharmeshvaja91 well, now logic seems incomplete: https://github.com/magento/magento2/pull/19503/files

**Comment by Dharmeshvaja91** [comment]
> @Dharmeshvaja91 well, now logic seems incomplete: https://github.com/magento/magento2/pull/19503/files

@orlangur can you please provide more info about "logic incomplete".

**Comment by orlangur** [comment]
@Dharmeshvaja91 you only set `isbothAddressSame` and never read it.

**Comment by Dharmeshvaja91** [comment]
> @Dharmeshvaja91 you only set `isbothAddressSame` and never read it.

@orlangur i have done as your suggested

**Comment by orlangur** [comment]
@Dharmeshvaja91 I didn't suggest anything as I don't understand the logic :)

`quote.billingAddress(quote.billingAddress());` how this code actually fixes the problem?

File: app/code/Magento/Checkout/view/frontend/web/js/view/billing-address.js [hunk 2]
```diff
@@ -127,11 +128,13 @@ function (
          */
         useShippingAddress: function () {
             if (this.isAddressSameAsShipping()) {
+                window.isbothAddressSame = true;
                 selectBillingAddress(quote.shippingAddress());
 
                 this.updateAddresses();
                 this.isAddressDetailsVisible(true);
             } else {
+                window.isbothAddressSame = false;
```
Comment (VladimirZaets) [inline app/code/Magento/Checkout/view/frontend/web/js/view/billing-address.js]: Hi @Dharmeshvaja91. Use global variables is bad practice. Please create local variable or set it to the class context.

File: app/code/Magento/Checkout/view/frontend/web/js/action/select-shipping-address.js [hunk 3]
```diff
@@ -12,6 +11,9 @@ define([
     'use strict';
 
     return function (shippingAddress) {
+        var address = null;
```
Comment (VladimirZaets) [inline app/code/Magento/Checkout/view/frontend/web/js/action/select-shipping-address.js]: Hi @Dharmeshvaja91. What the reason to set null to address variable and override it on the next line?

Comment (Dharmeshvaja91) [reply app/code/Magento/Checkout/view/frontend/web/js/action/select-shipping-address.js]: @VladimirZaets  ok than i directly override the address variable

File: app/code/Magento/Checkout/view/frontend/web/js/action/select-shipping-address.js [hunk 4]
```diff
@@ -12,6 +11,8 @@ define([
     'use strict';
 
     return function (shippingAddress) {
+        var address = quote.billingAddress();
         quote.shippingAddress(shippingAddress);
+        quote.billingAddress(address);
```
Comment (VladimirZaets) [inline app/code/Magento/Checkout/view/frontend/web/js/action/select-shipping-address.js]: Hi @Dharmeshvaja91. Could you please explain why you set the same value to the same observable variable. How it fix the issue?

**Comment by VladimirZaets** [comment]
@Dharmeshvaja91, I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for collaboration

**Comment by sidolov** [comment]
Hi @Dharmeshvaja91 , I'm closing this pull request because there is no activity since it was reopened. Please, push new changes and reopen pull request if you wish to continue. 
Thank you for the collaboration!

**Pull request closed without merging**